test/e2e/lib/helpers/broker_event_transformation_test_with_source_helper.go

```diff
@@ -0,0 +1,331 @@
+	// Tried with 45 seconds but the test has been quite flaky.
+	time.Sleep(90 * time.Second)
+	topicName, deleteTopic := lib.MakeTopicWithNameOrDie(t, topicName)
+	defer deleteTopic()
+
+	msg, err := client.WaitUntilJobDone(client.Namespace, targetName)
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Logf("Last term message => %s", msg)
+
+	if msg != "" {
+		out := &lib.TargetOutput{}
+		if err := json.Unmarshal([]byte(msg), out); err != nil {
+			t.Error(err)
+		}
+		if !out.Success {
+			// Log the output cloudauditlogssource pods.
+			if logs, err := client.LogsFor(client.Namespace, auditlogsName, lib.CloudAuditLogsSourceTypeMeta); err != nil {
+				t.Error(err)
+			} else {
+				t.Logf("cloudauditlogssource: %+v", logs)
+			}
+			// Log the output of the target job pods.
+			if logs, err := client.LogsFor(client.Namespace, targetName, lib.JobTypeMeta); err != nil {
+				t.Error(err)
+			} else {
+				t.Logf("job: %s\n", logs)
+			}
+			t.Fail()
+		}
+	}
+}
+
+func BrokerEventTransformationTestWithSchedulerSourceHelper(t *testing.T, client *lib.Client, authConfig lib.AuthConfig, brokerURL url.URL, brokerName string) {
+	data := "my test data"
+	targetName := "event-display"
+	sName := "scheduler-test"
+	// Create a target Job to receive the events.
+	lib.MakeSchedulerJobOrDie(client, data, targetName)
+	createTriggersAndKService(client, brokerName, targetName)
+
+	var url apis.URL = apis.URL(brokerURL)
+	// Just to make sure all resources are ready.
+	time.Sleep(5 * time.Second)
+
+	// Create the CloudSchedulerSource.
+	lib.MakeSchedulerOrDie(client, sName, data, targetName, authConfig.PubsubServiceAccount,
+		kngcptesting.WithCloudSchedulerSourceSinkURI(&url),
+	)
+
+	msg, err := client.WaitUntilJobDone(client.Namespace, targetName)
+	if err != nil {
+		t.Error(err)
+	}
+
+	t.Logf("Last termination message => %s", msg)
+	if msg != "" {
+		out := &lib.TargetOutput{}
+		if err := json.Unmarshal([]byte(msg), out); err != nil {
+			t.Error(err)
+		}
+		if !out.Success {
+			// Log the output of scheduler pods
+			if logs, err := client.LogsFor(client.Namespace, sName, lib.CloudSchedulerSourceTypeMeta); err != nil {
+				t.Error(err)
+			} else {
+				t.Logf("scheduler log: %+v", logs)
+			}
+
+			// Log the output of the target job pods
+			if logs, err := client.LogsFor(client.Namespace, targetName, lib.JobTypeMeta); err != nil {
+				t.Error(err)
+			} else {
+				t.Logf("addressable job: %+v", logs)
+			}
+			t.Fail()
+		}
+	}
+
+}
+
+func CreateKService(client *lib.Client) string {
+	kserviceName := helpers.AppendRandomString("kservice")
+	// Create the Knative Service.
+	kservice := resources.ReceiverKService(
+		kserviceName, client.Namespace)
+	client.CreateUnstructuredObjOrFail(kservice)
+	return kserviceName
+
+}
+
+func createTriggerWithKServiceSubscriber(client *lib.Client, brokerName, kserviceName string) {
+	dummyTriggerName := "dummy-broker-" + brokerName
```

Comment:
This trigger is not "dummy", maybe just call it kserviceTrigger

Reply:
I thought we call it dummy because it doesn't filter anything and it's used for the dummy event. That's the naming convention for both knative-eventing and knative-gcp

Reply:
we do not need to follow those. I would either rename it or add a comment, the name is really misleading

Reply:
I will rename it.